sources/osgShader/Compiler.js

```diff
@@ -627,21 +629,14 @@ define( [
 
             var samplerName = 'Texture' + unit.toString();
             var textureSampler = this.getVariable( samplerName );
-
             if ( textureSampler === undefined ) {
-
-                if ( texture.className() === 'Texture' ) {
-                    textureSampler = this.getOrCreateSampler( 'sampler2D', samplerName );
-                } else if ( texture.className() === 'TextureCubeMap' ) {
-                    textureSampler = this.getOrCreateSampler( 'samplerCube', samplerName );
-                } else if ( texture.className() === 'ShadowTexture' ) {
-                    textureSampler = this.getOrCreateSampler( 'sampler2D', samplerName );
-                    // return now to prevent creation of useless FragTexCoord
-                    //( shadow creates its own texcoord)
-                    return;
-                }
-
-
+                textureSampler = this.getOrCreateSampler( 'sampler2D', samplerName );
+            } else if ( texture.className() === 'TextureCubeMap' ) {
+                textureSampler = this.getOrCreateSampler( 'samplerCube', samplerName );
+            } else if ( texture.className() === 'ShadowTexture' ) {
```

Comment:
Instead of ``=== undefined``, you can just write ``!textureSampler`` (it's not like we are testing a string or a number)
Also, looks like it's broken. We can't declare a samplerCube anymore.... since it will fallback in the first branch.
And, why do we always test for ``getVariable`` when we only do a ``getOrCreate`` just after?

In sketchfab we don't use ``declareTexture`` and ``declareTextures`` because they declare too much nodes even when it's not needed.
Ping me back so that we can discuss about re-use of sketchab compiler code about texture management.